Nessie 0.49.0 breaks Iceberg NessieCatalog tests under Java 8 with an NPE in `TypeAnnotationParser.mapTypeAnnotations`

CI run under https://github.com/apache/iceberg/pull/6875 : https://github.com/apache/iceberg/actions/runs/4207727034/jobs/7302879619

Server-side exception from a local run with Java 8:
```
java.lang.NullPointerException
	at sun.reflect.annotation.TypeAnnotationParser.mapTypeAnnotations(TypeAnnotationParser.java:356)
	at sun.reflect.annotation.AnnotatedTypeFactory$AnnotatedTypeBaseImpl.<init>(AnnotatedTypeFactory.java:139)
	at sun.reflect.annotation.AnnotatedTypeFactory.buildAnnotatedType(AnnotatedTypeFactory.java:65)
	at sun.reflect.annotation.TypeAnnotationParser.buildAnnotatedTypes(TypeAnnotationParser.java:137)
	at java.lang.reflect.Executable.getAnnotatedParameterTypes(Executable.java:695)
	at java.lang.reflect.Parameter.getAnnotatedType(Parameter.java:237)
	at org.hibernate.validator.internal.properties.javabean.JavaBeanParameter.getAnnotatedType(JavaBeanParameter.java:56)
	at org.hibernate.validator.internal.metadata.provider.AnnotationMetaDataProvider.findTypeAnnotationConstraintsForExecutableParameter(AnnotationMetaDataProvider.java:732)
	at org.hibernate.validator.internal.metadata.provider.AnnotationMetaDataProvider.getParameterMetaData(AnnotationMetaDataProvider.java:431)
	at org.hibernate.validator.internal.metadata.provider.AnnotationMetaDataProvider.findExecutableMetaData(AnnotationMetaDataProvider.java:308)
	at org.hibernate.validator.internal.metadata.provider.AnnotationMetaDataProvider.getMetaData(AnnotationMetaDataProvider.java:292)
	at org.hibernate.validator.internal.metadata.provider.AnnotationMetaDataProvider.getMethodMetaData(AnnotationMetaDataProvider.java:279)
	at org.hibernate.validator.internal.metadata.provider.AnnotationMetaDataProvider.retrieveBeanConfiguration(AnnotationMetaDataProvider.java:130)
	at org.hibernate.validator.internal.metadata.provider.AnnotationMetaDataProvider.getBeanConfiguration(AnnotationMetaDataProvider.java:120)
	at org.hibernate.validator.internal.metadata.BeanMetaDataManagerImpl.getBeanConfigurationForHierarchy(BeanMetaDataManagerImpl.java:234)
	at org.hibernate.validator.internal.metadata.BeanMetaDataManagerImpl.createBeanMetaData(BeanMetaDataManagerImpl.java:201)
	at org.hibernate.validator.internal.metadata.BeanMetaDataManagerImpl.getBeanMetaData(BeanMetaDataManagerImpl.java:165)
	at org.hibernate.validator.internal.engine.ValidatorImpl.getConstraintsForClass(ValidatorImpl.java:316)
	at org.glassfish.jersey.server.validation.internal.DefaultConfiguredValidator.getConstraintsForClass(DefaultConfiguredValidator.java:100)
	at org.glassfish.jersey.server.validation.internal.DefaultConfiguredValidator.onValidate(DefaultConfiguredValidator.java:134)
	at org.glassfish.jersey.server.validation.internal.ValidationInterceptorExecutor.proceed(ValidationInterceptorExecutor.java:95)
	at org.glassfish.jersey.server.validation.internal.DefaultConfiguredValidator.validateResourceAndInputParams(DefaultConfiguredValidator.java:122)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(AbstractJavaResourceMethodDispatcher.java:124)
	at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$TypeOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:219)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:81)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:478)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:400)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:81)
	at org.glassfish.jersey.server.ServerRuntime$1.run(ServerRuntime.java:256)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:248)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:244)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:292)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:274)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:244)
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:265)
	at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:235)
	at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:684)
	at org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpContainer.service(GrizzlyHttpContainer.java:356)
	at org.glassfish.grizzly.http.server.HttpHandler$1.run(HttpHandler.java:200)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.doWork(AbstractThreadPool.java:569)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.run(AbstractThreadPool.java:549)
	at java.lang.Thread.run(Thread.java:750)
```


Obsolete (mistaken) previously mentioned exception:
```
java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[...]
	at java.lang.Thread.run(Thread.java:750)
Caused by: java.lang.NoSuchMethodError: com.google.common.base.Preconditions.checkState(ZLjava/lang/String;I)V
	at org.projectnessie.versioned.Key.check(Key.java:84)
[...]
```

--- a/clients/client/build.gradle.kts
+++ b/clients/client/build.gradle.kts
@@ -20,6 +20,7 @@ plugins {
   `maven-publish`
   signing
   `nessie-conventions`
+  alias(libs.plugins.annotations.stripper)
 }
 
 extra["maven.name"] = "Nessie - Client"
@@ -169,4 +170,11 @@ jacksonTestVersions.forEach { (jacksonVersion, reason) ->
     }
 
   jacksonTests { dependsOn(testTask, testTask8) }
+}
+
+annotationStripper {
+  registerDefault().configure {
+    annotationsToDrop("^jakarta[.].+".toRegex())
+    unmodifiedClassesForJavaVersion.set(11)
+  }
 }

--- a/gradle/libs.versions.toml
+++ b/gradle/libs.versions.toml
@@ -212,6 +212,7 @@ undertow-servlet = { module = "io.undertow:undertow-servlet", version.ref = "und
 weld-se-core = { module = "org.jboss.weld.se:weld-se-core", version = "3.1.9.Final" }
 
 [plugins]
+annotations-stripper = { id = "org.projectnessie.annotation-stripper", version = "0.1.0" }
 errorprone = { id = "net.ltgt.errorprone", version = "3.0.1" }
 gatling = { id = "io.gatling.gradle", version = "3.9.2" }
 idea-ext = { id = "org.jetbrains.gradle.plugin.idea-ext", version = "1.1.7" }

--- a/model/build.gradle.kts
+++ b/model/build.gradle.kts
@@ -28,6 +28,7 @@ plugins {
   `maven-publish`
   signing
   `nessie-conventions`
+  alias(libs.plugins.annotations.stripper)
 }
 
 apply<SmallryeOpenApiPlugin>()
@@ -87,3 +88,10 @@ val generateOpenApiSpec =
   }
 
 artifacts { add(openapiSource.name, file("src/main/resources/META-INF")) }
+
+annotationStripper {
+  registerDefault().configure {
+    annotationsToDrop("^jakarta[.].+".toRegex())
+    unmodifiedClassesForJavaVersion.set(11)
+  }
+}

--- a/servers/jax-rs/build.gradle.kts
+++ b/servers/jax-rs/build.gradle.kts
@@ -20,6 +20,7 @@ plugins {
   `maven-publish`
   signing
   `nessie-conventions`
+  alias(libs.plugins.annotations.stripper)
 }
 
 extra["maven.name"] = "Nessie - JAX-RS"
@@ -56,4 +57,11 @@ dependencies {
   api(platform(libs.jackson.bom))
   api(libs.jackson.databind)
   compileOnly(libs.jackson.annotations)
+}
+
+annotationStripper {
+  registerDefault().configure {
+    annotationsToDrop("^jakarta[.].+".toRegex())
+    unmodifiedClassesForJavaVersion.set(11)
+  }
 }

--- a/servers/rest-services/build.gradle.kts
+++ b/servers/rest-services/build.gradle.kts
@@ -20,6 +20,7 @@ plugins {
   `maven-publish`
   signing
   `nessie-conventions`
+  alias(libs.plugins.annotations.stripper)
 }
 
 extra["maven.name"] = "Nessie - REST Services"
@@ -60,4 +61,11 @@ dependencies {
   testImplementation(platform(libs.junit.bom))
   testImplementation(libs.bundles.junit.testing)
   testRuntimeOnly(libs.junit.jupiter.engine)
+}
+
+annotationStripper {
+  registerDefault().configure {
+    annotationsToDrop("^jakarta[.].+".toRegex())
+    unmodifiedClassesForJavaVersion.set(11)
+  }
 }

--- a/servers/services/build.gradle.kts
+++ b/servers/services/build.gradle.kts
@@ -22,6 +22,7 @@ plugins {
   `maven-publish`
   signing
   `nessie-conventions`
+  alias(libs.plugins.annotations.stripper)
 }
 
 extra["maven.name"] = "Nessie - Services"
@@ -93,4 +94,11 @@ dependencies {
 // Issue w/ testcontainers/podman in GH workflows :(
 if (Os.isFamily(Os.FAMILY_MAC) && System.getenv("CI") != null) {
   tasks.withType<Test>().configureEach { this.enabled = false }
+}
+
+annotationStripper {
+  registerDefault().configure {
+    annotationsToDrop("^jakarta[.].+".toRegex())
+    unmodifiedClassesForJavaVersion.set(11)
+  }
 }

--- a/servers/store/build.gradle.kts
+++ b/servers/store/build.gradle.kts
@@ -20,6 +20,7 @@ plugins {
   `maven-publish`
   signing
   `nessie-conventions`
+  alias(libs.plugins.annotations.stripper)
 }
 
 extra["maven.name"] = "Nessie - Server - Store"
@@ -49,4 +50,11 @@ dependencies {
   testImplementation(platform(libs.junit.bom))
   testImplementation(libs.bundles.junit.testing)
   testRuntimeOnly(libs.junit.jupiter.engine)
+}
+
+annotationStripper {
+  registerDefault().configure {
+    annotationsToDrop("^jakarta[.].+".toRegex())
+    unmodifiedClassesForJavaVersion.set(11)
+  }
 }

